### DISCUSSION TITLE
Subscriber getting stopped after some time while subscribing to real time continuous data

### DISCUSSION THREAD

**Discussion by Saurabh-Sinalkar** [OpenDDS/OpenDDS#3894] [Q&A] [answered]
We have a publisher subscriber opendds application on which we are sending a single topic real time continuous data.
The discovery and transport mode which we are using is rtps_udp and all the default qos policies are being used.
The publisher and subscriber works correctly at the beginning but after some time (approx 10k) data samples  the subscriber is getting stopped while the publisher is still continuously sending the data. 
The error logs on the subscriber side shows the writer becoming dead followed by on liveliness and on subscription matched events getting called as attached in the screenshot.
Can you please let us know why is the issue occuring?
![error2](https://user-images.githubusercontent.com/105099134/203246578-01ada141-fbc8-4c73-89cf-175def025735.JPG)
![error1](https://user-images.githubusercontent.com/105099134/203246588-54c1bce5-ba4a-4d1d-a589-c1634de15a4e.JPG)

**Comment by mitza-oci** [comment, marked as answer]
Which version of OpenDDS?

**Reply by Saurabh-Sinalkar** [reply]
OpenDDS version which we are using is 3.13

**Reply by mitza-oci** [reply]
Do see the same problem with the latest release?

**Reply by Saurabh-Sinalkar** [reply]
Yes it occurs with the latest version as well, it can be reproduced in the example messenger code by increasing the topic count to some high value.

**Reply by simpsont-oci** [reply]
There is only one topic in the messenger example. Are you referring to the message count? If so, are you increasing the expected count in the subscriber as well? You may also need to increase the duration values use in both the subscriber and publisher.

If you can post a patch file of your changes, that would be helpful to see the exact issue you're describing.

**Reply by Saurabh-Sinalkar** [reply]
Yes we are referring to the message count in publisher code in the for loop.
We are unable to understand where is the the expected count in subscriber & how to increase and also where and how to increase the duration value in publisher and subscriber code.
Attaching the code snippet for the message count change in publisher code we made and also the error screenshot wherein all the samples are not been received.
![MicrosoftTeams-image (1)](https://user-images.githubusercontent.com/105099134/203805848-22e1a1d4-475a-44d6-96f9-9db14338070e.png)
![MicrosoftTeams-image](https://user-images.githubusercontent.com/105099134/203805852-86f21c07-f267-4247-9973-5c420cb38d06.png)

**Reply by mitza-oci** [reply]
Please use a pull request to add a test to the repository
